LiskHQ/lisk-hub
# Implement form step 1/3 of send BTC - Closes #1889

### What issue have I solved?

#1889

### How have I implemented/fixed it?

- Removed unnecessary HOC from the form;
- Updated form to get activeToken from store, if `localStorage` has the 'btc' key set;
- Adjusted bookmarks to only show bookmarks with same token as active token;
- Adjusted address validation to be token based;
- Adjusted amount validation to be based on the balance of the active token;
- Implemented Dynamic Fee for BTC;
- Created Selector component to be easier to reuse on the future;

**Know Issues**
- At the moment we don't have an API to provide the "estimated processing time", so it is not displayed.
- We currently don't have the BTC/LSK switcher on the send, we are considering that the token is already set before going to the send page.

### How has this been tested?

- Go to `/wallet` and go to send page;
- It should be exactly the same as before;
- Set `localStorage.setItem('btc', true);`;
- Update the active token to 'BTC':
  - `$r.store.dispatch({type:'SETTINGS_UPDATE_TOKEN', data: { token: { active: 'BTC', list: {BTC: true, LSK: true} }}})` Note that React dev tools have to be active for $r to be defined.
  - Or dispatching `{type:'SETTINGS_UPDATE_TOKEN', data: { token: { active: 'BTC', list: {BTC: true, LSK: true} }}}` through Redux DevTools.
- The page should show the recipient and amount fields, and after inputing the amount, should fetch the dynamic Fee, and display it on the page.

### Review checklist
- The PR follows our [Test guide](/LiskHQ/lisk-hub/blob/development/docs/TEST_GUIDE.md)
- The PR follows our [CSS guide](/LiskHQ/lisk-hub/blob/development/docs/CSS_GUIDE.md)
